From 1fb14458247efa35492973ff7778788c3cb0af41 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 9 Oct 2025 17:01:53 +0200
Subject: [PATCH] arm64: dts: st: add cryp2 node for STM32MP23

Add cryp2 node in STM32MP23 DT for SoC type C and F.

Change-Id: I3f5656168f577a61111880d94a459dbb0b60a838
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/498121
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp23xc.dtsi | 14 ++++++++++++++
 arch/arm64/boot/dts/st/stm32mp23xf.dtsi | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp23xc.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp23xc.dtsi
@@ -18,4 +18,18 @@
 		power-domains = <&d1_pd>;
 		status = "disabled";
 	};
+
+	cryp2: crypto@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x1000>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		dmas = <&hpdma 140 0x40 0x3021>,
+		       <&hpdma 141 0x43 0x3012>;
+		dma-names = "in", "out";
+		access-controllers = <&rifsc 97>;
+		power-domains = <&d1_pd>;
+		status = "disabled";
+	};
 };
--- a/arch/arm64/boot/dts/st/stm32mp23xf.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp23xf.dtsi
@@ -18,4 +18,18 @@
 		power-domains = <&d1_pd>;
 		status = "disabled";
 	};
+
+	cryp2: crypto@42040000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42040000 0x1000>;
+		interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP2>;
+		resets = <&rcc CRYP2_R>;
+		dmas = <&hpdma 140 0x40 0x3021>,
+		       <&hpdma 141 0x43 0x3012>;
+		dma-names = "in", "out";
+		access-controllers = <&rifsc 97>;
+		power-domains = <&d1_pd>;
+		status = "disabled";
+	};
 };
